connman/src/service.c

```diff
@@ -9176,8 +9176,13 @@ void connman_service_create_ip4config(struct connman_service *service,
 {
 	DBG("ipv4 %p", service->ipconfig_ipv4);
 
-	if (service->ipconfig_ipv4)
+	if (service->ipconfig_ipv4) {
+		if (connman_ipconfig_get_index(service->ipconfig_ipv4) != index)
+			__connman_ipconfig_set_index(service->ipconfig_ipv4,
```

Comment:
Is there some benefit for condtitional set_index()? In other commit it was done unconditionally + set_index() is basically a logical nop if there is no index change.

Reply:
Well, maybe not. I'm just wondering if this would possibly break anything in some weird CLAT-scenario. However, the index should always be set to match the interface when the ipconfig is to be created. In many cases it may be set later on but that would be an explicit set and this just simply ensures that the index is always matched with the interface.

Reply:
Removed that bit to skip the check. Yeah, it wasn't necessary. Overthinking again.